src/inflators/loop-animation.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  addComponent,
+  hasComponent
+} from "bitecs";
+import {
+  AnimationAction,
+  AnimationClip,
+  AnimationMixer,
+  LoopRepeat
+} from "three";
+import {
+  LoopAnimation,
+  LoopAnimationData,
+  MixerAnimatable,
+  MixerAnimatableData,
+  Object3DTag
+} from "../bit-components";
+import { HubsWorld } from "../app";
+
+// TODO: Type definition should be done in bit-components?
+type MixerAnimatableData = Map<number, {
+  mixer: AnimationMixer,
+  animations: Array<AnimationClip>
+}>;
+type LoopAnimationSata = Map<number, Array<AnimationAction>>;
+
+export type LoopAnimationParams = {
+  activeClipIndex: number;
+  paused: boolean;
+  startOffset: number;
+  timeScale: number;
+  // TODO: Do we need to keep supporting the following two params?
+  // DEPRECATED: Use activeClipIndex instead since animation
+  //             names are not unique
+  clip?: string;
+  // Support for Spoke->Hubs activeClipIndices struct
+  activeClipIndices?: Array<number>;
+};
+
+const DEFAULTS = {
+  activeClipIndex: 0,
+  paused: false,
+  startOffset: 0,
+  timeScale: 1.0
+};
+
+const getActiveClips = (animations: Array<AnimationClip>, params: LoopAnimationParams) : Array<AnimationClip> => {
+  if (params.activeClipIndices !== undefined &&
+      params.activeClipIndices.length > 0) {
+    return params.activeClipIndices.map(index => animations[index]);
+  } else if (params.clip !== undefined &&
+             params.clip !== "") {
+    const activeClips = [];
+    const clipNames = params.clip!.split(",");
+    for (let i = 0; i < clipNames.length; i++) {
+      const clipName = clipNames[i];
+      const foundClip = animations.find((clip: AnimationClip) => {
+        return clip.name === clipName;
+      });
+      if (foundClip) {
+        activeClips.push(foundClip);
+      } else {
+        console.warn(`Could not find animation names '${clipName}' in `, animations);
+      }
+    }
+    return activeClips;
+  } else {
+    return [animations[params.activeClipIndex]];
+  }
+};
+
+export function inflateLoopAnimation(world: HubsWorld, eid: number, params: LoopAnimationParams) : number {
+  // Is it possible to guarantee that LoopAnimation
```

Comment:
No it's not. It's invalid to try and get the Object3D in an inflator. An inflator may be the thing that ends up creating the Object3D for an entity. Needing to get the object is a good sign that this code should be in an enter query instead. 

inflators should just set up the component data and (if they are a component managing *the* Object3D for an entity) create the Object3D and call `addObject3DComponent`. We definitely need to make sure to call this out clearly in the docs since I suspect it will be a repeated point of confusion.

Reply:
Related to Takahiro's comment, is there any reason why we are adding the Object3D here the last thing?:
https://github.com/mozilla/hubs/blob/a183e23e609f86db0612ccfb0b3b8e495625087a/src/inflators/model.tsx#L73-L92

If the Object3D entity was added first thing the component could have access to the Object3D in case it needs certain properties for any sort of initialization (scale, rotation, etc) instead of needing to implement a system/query just for that.